### ISSUE TITLE
Lambda Timeout waiting for function to be ready for invocation

### ISSUE THREAD

**Issue by nathankurt** [artilleryio/artillery#1772]
Version info: 
```
VERSION INFO:

Artillery: 2.0.0-28
Node.js:   v16.15.1
OS:        linux
```

Running this command: 
```
artillery run --platform aws:lambda --platform-opt region=us-east-1 --count 25 blitz.yml
```

I expected to see this happen:

When I run `artillery run blitz.yml` the command works as expected, and the load test is run, when i run the command above, i expect the same thing to happen just via the lambda. 

Instead, this happened:

I get an error after the build phase in AWS Codebuild where we are trying to run this. I didn't see a way to increase the timeout anywhere but that seems like it might be needed. 
```
Creating AWS Lambda function...
- Bundling test data
  - blitz.yml
- Installing dependencies
- Creating zip package
Preparing AWS environment...
 - Lambda function: artilleryio-[blahblahblah]
 - Region: us-east-1
 - AWS account: [ACT NUMBER]
⠋ /usr/local/lib/node_modules/artillery/lib/platform/aws-lambda/index.js:420
      throw new Error(`Timeout waiting for lambda function to be ready for invocation`);
            ^

Error: Timeout waiting for lambda function to be ready for invocation
    at PlatformLambda.runWorker (/usr/local/lib/node_modules/artillery/lib/platform/aws-lambda/index.js:420:13)
    at async Launcher.run (/usr/local/lib/node_modules/artillery/lib/launch-platform.js:407:7)
```

Files being used:

buildspec.yaml file 
```yaml
version: '0.2'
env:
  shell: /bin/bash
  variables:
    ENVIRONMENT: development
phases:
  install:
    runtime-versions:
      nodejs: 16
  pre_build:
    commands:
    - echo "Starting the build"
    - npm install -g artillery
    - cd stacks/artillery
    - pwd
    - ls
  build:
    commands:
    - artillery run --platform aws:lambda --platform-opt region=us-east-1 --count
      25 blitz.yml
  post_build:
    commands:
    - echo Build completed on `date`

```

**Comment by hassy** [comment]
hi @nathankurt 👋 are you using `blitz.yml` from this blog post? https://www.artillery.io/blog/this-load-test-cost-us-how-much

**Comment by nathankurt** [comment]
Hi @hassy ! We are using that blog post with a different but very similar blitz.yaml file that hits our internal dev site instead

**Comment by hassy** [comment]
Alrighty, thank you. I can't reproduce it with the script from the blog post, even after I've added extra `get` steps and a `config.ensure` section. Same region, same count as in your command.

The timeout is currently hardcoded as 2 minutes, which *should* be plenty of time for AWS to prepare the Lambda function. Does this error happen for you consistently every time you try to run the test?

**Comment by nathankurt** [comment]
yep, this happens every time so far. I just tried switching to your exact yaml file in the blog post, and the same error happens. 
```
sh-4.2# artillery run --platform aws:lambda --platform-opt region=us-east-1 --count 25 blitz.yml
NOTE: AWS Lambda support is experimental. Not all Artillery features work yet.
For details please see https://docs.art/aws-lambda

λ Creating AWS Lambda function...
- Bundling test data
  - blitz.yml
- Installing dependencies
- Creating zip package
Preparing AWS environment...
 - Lambda function: artilleryio-53ee73c2-1247-4b90-bdde-77a88e181470
 - Region: us-east-1
 - AWS account: 810605503585
⠸ /usr/local/lib/node_modules/artillery/lib/platform/aws-lambda/index.js:420
      throw new Error(`Timeout waiting for lambda function to be ready for invocation`);
            ^

Error: Timeout waiting for lambda function to be ready for invocation
    at PlatformLambda.runWorker (/usr/local/lib/node_modules/artillery/lib/platform/aws-lambda/index.js:420:13)
    at async Launcher.run (/usr/local/lib/node_modules/artillery/lib/launch-platform.js:407:7)
sh-4.2#
```

Note that this is inside AWS CodeBuild that we're running it so that probably makes a difference, but i can't see how it makes that much of a difference.

**Comment by nathankurt** [comment]
So tried cloning the repo and running from source. Updated the line right [here](https://github.com/artilleryio/artillery/blob/203d0b868267f66ec36f4cfd71cbbd2d0872ba60/packages/artillery/lib/platform/aws-lambda/index.js#L400) to increase the timeout to 300 and also 600 instead of 120. still failed. But when i ran the command `aws lambda get-function --function-name artilleryio-36afc8c0-ce8e-46c9-b390-81779c5fcf2e --query 'Configuration.[State, LastUpdateStatus]'` while the command was running (that's the name of the latest lambda) i got a response back that the lambda was active already. 

But when i tried to print out the state, nothing was printing out. Realized it didn't have the permission to get that state, so I needed to add `lambda:GetFunctionConfiguration` to the IAM Role which wasn't listed in the blog post. 

Now it works! Thanks for your help!
